### PR TITLE
[CORE-1856] Correct Proto compatibility

### PR DESCRIPTION
**Pull request by robert-uhl**
* Properly reserve old field IDs and names.
* Rework message flow in order to support UX.
   * User views and edits entire cluster defaults *as JSON*, including the default CreatePipelineRequest.
   * Cluster defaults JSON is stored unchanged in the database using an internal wrapper rather than as a ClusterDefaults message. The wrapper is wire-compatible with ClusterDefaults in order to be compatible with v2.7.

### CODE REVIEW

File: src/internal/ppsdb/ppsdb.proto [hunk 1]
```diff
@@ -0,0 +1,12 @@
+syntax = "proto3";
+
+package pps_v2;
+option go_package = "github.com/pachyderm/pachyderm/v2/src/internal/ppsdb";
+
+import "pps/pps.proto";
+
+message ClusterDefaultsWrapper {
+  reserved 1, 2;
+  reserved "details_json", "effective_details_json";
```
Comment (acohen4) [inline src/internal/ppsdb/ppsdb.proto:10]: why reserve this?

File: .golangci.yml [hunk 2]
```diff
@@ -48,6 +48,7 @@ linters-settings:
             - github.com/pachyderm/pachyderm/v2/src/internal/errors.Invoke1
             - google.golang.org/grpc/status.Error
             - google.golang.org/grpc/status.Errorf
+            - "(*google.golang.org/grpc/internal/status.Status).Err"
```
Comment (acohen4) [inline .golangci.yml:51]: why is this necessary?

**Review by acohen4 — APPROVED** [review]
LGTM, a couple Qs

File: src/internal/ppsdb/ppsdb.proto (hunk 1 above)
Comment (robert-uhl) [reply src/internal/ppsdb/ppsdb.proto:10]: Because in v2.7, this was a message with field 1 named `details_json` & field 2 named `effective_details_json`; it was written to the database but never actually used. Reserving the fields lets marshalling & unmarshalling work without an error (and they can be ignored because they weren’t actually used). It’s purely a compatibility measure.

File: .golangci.yml (hunk 2 above)
Comment (robert-uhl) [reply .golangci.yml:51]: Allows getting rid of `//nolint:wrapcheck` when returning a gRPC status error.

Not strictly necessary in this PR, but it’s about to be used in the follow-ons.
